vars/pnc.groovy

```diff
@@ -0,0 +1,96 @@
+import groovy.json.JsonSlurper
+
+/**
+ * Query the PNC api. The base URL must be configured in PNC_API_URL environment variable.
+ * @param endpoint PNC endpoint to invoke
+ * @param params key-value map of additional parameters
+ * @param pageIndex page to retrieve
+ */
+def queryPNC(String endpoint, Map params, int pageIndex=0) {
+    def queryUrl = "${env.PNC_API_URL}/${endpoint}?pageIndex=${pageIndex}&pageSize=200&" + util.serializeQueryParams(params)
+    println "[INFO] Querying PNC ${queryUrl}"
+
+    def response = sh( [script: "curl -s -X GET \"Accept: application/json\" \"${queryUrl}\"", returnStdout: true] )
+    // println "[DEBUG] Received from PNC: ${response}"
+    return new JsonSlurper().parseText(response as String)
+}
+
+/**
+ * Returns the number of pages returned by an API call
+ */
+int getPagesNumber(String endpoint, Map params) {
+    return queryPNC(endpoint, params).totalPages as int
+}
+
+/**
+ * Retrieve all the product's milestones from PNC
+ * @param productId id of the product in PNC
+ */
+def getAllMilestonesForProduct(String productId) {
+    println "[INFO] Getting all milestones for product ${productId}"
+    return queryPNC("products/${productId}/versions", [q: ""])
+}
+
+/**
+ * Retrieve the latest (i.e., current) product's milestone from PNC
+ * @param productId id of the product in PNC
+ */
+def getCurrentMilestoneForProduct(String productId) {
+    println "[INFO] Getting current milestone for product ${productId}"
+    def result = getAllMilestonesForProduct(productId)
+    return result.content.last().currentProductMilestone
+}
+
+/**
+ * Retrieve the PNC identifier of a specific product's milestone
+ * @param productId id of the product in PNC
+ * @param milestone
+ */
+def getMilestoneId(String productId, String milestone) {
+    println "[INFO] Getting ${milestone} milestone id for product ${productId}"
+
+    // TODO  optimize
+    def found = getAllMilestonesForProduct(productId)
```

Comment:
what about? :thinking: 
```
getAllMilestonesForProduct(productId)
            .collect { it.content.productMilestones }
            .collect { it.value }
            .find { it.version == milestone }
            .first()
```

Reply:
I guess this should be the correct one:
```
getAllMilestonesForProduct(productId).content.productMilestones.collectMany { it.values() }.find { it.version == milestone }
```
wdyt?